### PR TITLE
Adding setting for OE expansion timeout

### LINKED ISSUE #19276: Setting for Object Explorer timeout
We should add a setting in Object Explorer to let users add a custom timeout for expansion instead of defaulting to 45 seconds.

### PR DESCRIPTION
**Pull request by aasimkhan30**
This PR fixes #19276 

image

### CODE REVIEW

File: extensions/mssql/package.json [hunk 1]
```diff
@@ -371,6 +371,11 @@
           "type": "boolean",
           "default": false,
           "description": "%mssql.objectExplorer.groupBySchema%"
+        },
+        "mssql.objectExplorer.ExpandTimeout": {
```
Comment (alanrenmsft) [inline extensions/mssql/package.json:375]: expandTimeout. E should be lower case

**Review by alanrenmsft — APPROVED** [review]

**Comment by lewis-sanchez** [comment]
I'm sure this value gets read somewhere, so whatever triggers the timeout knows to wait the time specified by the user, but it's not clear in the PR.

**Review by lewis-sanchez — APPROVED** [review]

File: extensions/mssql/package.json [hunk 2]
```diff
@@ -371,6 +371,11 @@
           "type": "boolean",
           "default": false,
           "description": "%mssql.objectExplorer.groupBySchema%"
+        },
+        "mssql.objectExplorer.expandTimeout": {
+          "type": "number",
+          "default": 45,
+          "description": "%mssql.objectExplorer.expandTimeout%"
```
Comment (Charles-Gagnon) [inline extensions/mssql/package.json:378]: Is this already set up to be consumed in STS?

File: extensions/mssql/package.json [hunk 3]
```diff
@@ -371,6 +371,11 @@
           "type": "boolean",
           "default": false,
           "description": "%mssql.objectExplorer.groupBySchema%"
+        },
+        "mssql.objectExplorer.expandTimeout": {
+          "type": "number",
+          "default": 45,
```
Comment (Charles-Gagnon) [inline extensions/mssql/package.json:377]: Add minimum of 1

File: extensions/mssql/package.json (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/mssql/package.json:378]: Guess so, @llali I don't suppose you know why this doesn't appear to exist? Did it never get added on the ADS side?

https://github.com/microsoft/sqltoolsservice/pull/363

**Comment by Charles-Gagnon** [comment]
Are you going to add this to VS Code MSSQL as well?

File: extensions/mssql/package.json (hunk 2 above)
Comment (aasimkhan30) [reply extensions/mssql/package.json:378]: Yeah, OE setting is already using this in STS. https://github.com/microsoft/sqltoolsservice/blob/c26a2aea142524c9dec4625448fe0f1ff2b2fcde/src/Microsoft.SqlTools.ServiceLayer/SqlContext/ObjectExplorerSettings.cs#L34

**Review by kburtram — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**